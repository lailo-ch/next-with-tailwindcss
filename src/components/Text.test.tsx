import * as React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Text from './Text'

describe('Text', () => {
  it('matches h1 snapshot', () => {
    const wrapper = shallow(<Text h1>Header 1</Text>)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('matches h2 snapshot', () => {
    const wrapper = shallow(<Text h2>Header 2</Text>)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('matches h3 snapshot', () => {
    const wrapper = shallow(<Text h3>Header 3</Text>)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('matches h4 snapshot', () => {
    const wrapper = shallow(<Text h4>Header 4</Text>)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('matches h5 snapshot', () => {
    const wrapper = shallow(<Text h5>Header 5</Text>)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('matches h6 snapshot', () => {
    const wrapper = shallow(<Text h6>Header 6</Text>)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('matches default text snapshot', () => {
    const wrapper = shallow(<Text>This is a paragraph</Text>)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('matches small snapshot', () => {
    const wrapper = shallow(<Text small>Small text</Text>)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('matches bold snapshot', () => {
    const wrapper = shallow(<Text bold>Bold text</Text>)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
