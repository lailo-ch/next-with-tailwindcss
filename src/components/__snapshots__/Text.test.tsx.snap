// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text matches bold snapshot 1`] = `
<p
  className="font-light leading-relaxed font-semibold text-base text-gray-700 dark:text-gray-300"
>
  Bold text
</p>
`;

exports[`Text matches default text snapshot 1`] = `
<p
  className="font-light leading-relaxed text-base text-gray-700 dark:text-gray-300"
>
  This is a paragraph
</p>
`;

exports[`Text matches h1 snapshot 1`] = `
<h1
  className="text-5xl text-gray-900 dark:text-gray-100 font-black"
>
  Header 1
</h1>
`;

exports[`Text matches h2 snapshot 1`] = `
<h2
  className="text-4xl text-gray-900 dark:text-gray-100 font-black"
>
  Header 2
</h2>
`;

exports[`Text matches h3 snapshot 1`] = `
<h3
  className="text-3xl text-gray-900 dark:text-gray-100 font-black"
>
  Header 3
</h3>
`;

exports[`Text matches h4 snapshot 1`] = `
<h4
  className="text-2xl text-gray-900 dark:text-gray-100 font-black"
>
  Header 4
</h4>
`;

exports[`Text matches h5 snapshot 1`] = `
<h5
  className="text-xl text-gray-900 dark:text-gray-100 font-black"
>
  Header 5
</h5>
`;

exports[`Text matches h6 snapshot 1`] = `
<h6
  className="text-lg text-gray-900 dark:text-gray-100 font-black"
>
  Header 6
</h6>
`;

exports[`Text matches small snapshot 1`] = `
<small
  className="text-xs text-gray-700 dark:text-gray-300"
>
  Small text
</small>
`;
